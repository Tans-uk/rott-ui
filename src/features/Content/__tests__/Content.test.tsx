import {Content} from '@features'
import {render} from '@utils'

describe('Content -> Custom Component', () => {
  const testId = {
    keyboardAvoidingViewTestId: 'keyboard-avoiding-view-test-id',
  }

  it('ilk render anında snapshot ile eşleşmeli', () => {
    const rendered = render(<Content />)

    expect(rendered).toMatchSnapshot()
  })

  it('keyboard avoiding view varsayilan ekranda olmamali', () => {
    const {keyboardAvoidingViewTestId} = testId
    const {queryByTestId} = render(<Content />)

    const keyboardAvoidingViewElement = queryByTestId(keyboardAvoidingViewTestId)

    expect(keyboardAvoidingViewElement).not.toBeTruthy()
  })

  it('keyboardAvoidingView verildiginde renderlanmali', () => {
    const {keyboardAvoidingViewTestId} = testId
    const {getByTestId} = render(<Content keyboardAvoidingView />)

    const keyboardAvoidingViewElement = getByTestId(keyboardAvoidingViewTestId)

    expect(keyboardAvoidingViewElement).toBeTruthy()
  })
})
