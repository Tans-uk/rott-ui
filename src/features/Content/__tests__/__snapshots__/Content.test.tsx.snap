// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content -> Custom Component ilk render anında snapshot ile eşleşmeli 1`] = `
<KeyboardProvider
  enabled={true}
>
  <View
    flex={1}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        {
          "flex": undefined,
          "flexDirection": "column",
          "paddingBottom": 0,
          "paddingHorizontal": 46,
        }
      }
    />
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignSelf": "center",
            "elevation": 999999,
            "flex": 0,
            "maxWidth": "100%",
            "position": "absolute",
            "zIndex": 999999,
          },
          {
            "flexDirection": "column-reverse",
            "justifyContent": "flex-start",
            "top": 30,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          {
            "bottom": 0,
          }
        }
      >
        <RCTSafeAreaView />
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignSelf": "center",
            "elevation": 999999,
            "flex": 0,
            "maxWidth": "100%",
            "position": "absolute",
            "zIndex": 999999,
          },
          {
            "bottom": 40,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          {
            "bottom": 0,
          }
        }
      >
        <RCTSafeAreaView />
      </View>
    </View>
  </View>
</KeyboardProvider>
`;
